### DISCUSSION TITLE
For Automatic Periodic Boundary Detection, can it use AuxVariables? It is only Variables System.

### DISCUSSION THREAD

**Discussion by snugook1108** [idaholab/moose#21354] [Q&A General] [answered]
## Reason
For Automatic Periodic Boundary Detection, can it use the AuxVariables? It is only for the Variables System. 

## Design
```
[BCs]
  [./Periodic]
    [./all]
      variable = u
      auto_direction = 'x y'
    [../]
  [../]
[]
```
The variable is u so if it is possible, the variable can use  AuxVariables?

**Comment by GiudGiud** [comment, marked as answer]
We dont set boundary conditions to auxvariables. AuxVariables are computed by AuxKernels, they are not solved for. If you want to make a periodic one, you can code this into the AuxKernel definition

btw we use the Discussions forum for such questions, not issues
